packages/core/src/shared/vscode/setContext.ts

```diff
@@ -10,19 +10,21 @@ import * as vscode from 'vscode'
  *
  * New keys must start with "aws." or "amazonq.".
  */
-type contextKey =
+export type contextKey =
     | 'aws.isDevMode'
     | 'aws.isSageMaker'
     | 'aws.isWebExtHost'
     | 'aws.isInternalUser'
     | 'aws.amazonq.showLoginView'
+    | 'aws.amazonq.notifications.show'
     | 'aws.codecatalyst.connected'
     | 'aws.codewhisperer.connected'
     | 'aws.codewhisperer.connectionExpired'
     | 'aws.codewhisperer.tutorial.workInProgress'
     | 'aws.explorer.showAuthView'
     | 'aws.toolkit.amazonq.dismissed'
     | 'aws.toolkit.amazonqInstall.dismissed'
+    | 'aws.toolkit.notifications.show'
```

Comment:
Unlike globalState (which definitely is local to each extension), I haven't confirmed if setContext keys are "local" to each extension. So it may be required to define a q-specific and toolkit-specific key, as you've done here. However, Claude 3 claims that setContext keys are isolated per-extension:


> extension has its own namespace for the `setContext` command, ensuring that there is no conflict between different extensions defining context keys with the same name.
> 
> When an extension defines a context key using the `setContext` command, the key is automatically prefixed with the extension's unique identifier (the `publisher.name` from the extension's `package.json` file). This namespacing mechanism ensures that context keys defined by different extensions are kept separate and do not conflict with each other.
> 
> For example, if extension A defines a context key called `myKey` and extension B also defines a context key called `myKey`, the actual context keys stored in Visual Studio Code would be:
> 
> - `extension-a.myKey` (for the key defined by extension A)
> - `extension-b.myKey` (for the key defined by extension B)
> 
> This namespacing applies not only to context keys defined by `setContext` but also to context keys referenced in other parts of an extension, such as in `when` clauses for command visibility or in `package.json` directives.

Reply:
Tested - they are shared. 
 Claude is a liar 😡